rq/worker_pool.py

```diff
@@ -177,26 +182,74 @@ def start_worker(
         logging_level: str = 'INFO',
+        with_scheduler: bool = True,
     ):
         """
         Starts a worker and adds the data to worker_datas.
         * sleep: waits for X seconds before creating worker, for testing purposes
         """
         name = uuid4().hex
-        process = self.get_worker_process(name, burst=burst, _sleep=_sleep, logging_level=logging_level)
+        process = self.get_worker_process(
+            name, burst=burst, _sleep=_sleep, logging_level=logging_level, with_scheduler=with_scheduler
+        )
         process.start()
         worker_data = WorkerData(name=name, pid=process.pid, process=process)  # type: ignore
         self.worker_dict[name] = worker_data
         self.log.debug('Spawned worker: %s with PID %d', name, process.pid)
 
-    def start_workers(self, burst: bool = True, _sleep: float = 0, logging_level: str = 'INFO'):
+    def start_workers(
+        self,
+        burst: bool = True,
+        _sleep: float = 0,
+        logging_level: str = 'INFO',
+        with_scheduler: bool = False,
+    ):
         """
         Run the workers
         * sleep: waits for X seconds before creating worker, only for testing purposes
         """
         self.log.debug(f'Spawning {self.num_workers} workers')
         for i in range(self.num_workers):
-            self.start_worker(i + 1, burst=burst, _sleep=_sleep, logging_level=logging_level)
+            self.start_worker(
+                i + 1, burst=burst, _sleep=_sleep, logging_level=logging_level, with_scheduler=with_scheduler
+            )
+
+    def _start_scheduler(self, burst: bool = False, logging_level: Union[str, int] = 'INFO') -> None:
+        """Starts a single scheduler process for the pool's queues.
+
+        If burst is True, enqueue scheduled jobs once and return.
+        Otherwise, acquire locks and start the scheduler process.
+        """
+        self.scheduler = RQScheduler(
+            self.queues,
+            connection=self.connection,
+            logging_level=logging_level,
+            date_format=DEFAULT_LOGGING_DATE_FORMAT,
+            log_format=DEFAULT_LOGGING_FORMAT,
+            serializer=self.serializer,
+        )
+        self.scheduler.acquire_locks()
+        if self.scheduler.acquired_locks:
+            if burst:
+                self.scheduler.enqueue_scheduled_jobs()
+                self.scheduler.release_locks()
+            else:
+                self.scheduler.start()
+
+    def stop_scheduler(self) -> None:
+        """Ensure the scheduler process is stopped and joined."""
+        scheduler = self.scheduler
+        if scheduler is not None:
+            proc = scheduler._process
+            if proc is not None:
+                pid = getattr(proc, 'pid', None)
+                try:
+                    if pid:
+                        os.kill(pid, signal.SIGTERM)
+                except OSError:
+                    pass
+                finally:
+                    proc.join()
```

Comment:
Thanks for the PR. In general, I think too much code is added for this simple feature. At the very least, I think we should think about unifying the `start_scheduler()` and `stop_scheduler()` code blocks with the same functionality in `Worker`.